pkg/operator/status.go

```diff
@@ -29,7 +29,7 @@ func (optr *Operator) syncVersion() error {
 	}
 
 	// keep the old version and progressing if we fail progressing
-	if cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorProgressing) && cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorDegraded) {
+	if cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorProgressing) || cov1helpers.IsStatusConditionTrue(co.Status.Conditions, configv1.OperatorDegraded) {
```

Comment:
debugging with Alex. This `||` makes it impossing to flip from progressing=True to False because we now never report the new version to the cluster object.

Reply:
in other words, there's no way right now to flip from progressing=true & old version to progressing=false & new version reported here